source/Octopus.Tentacle.Tests/Client/RpcCallExecutorFixture.cs

```diff
@@ -0,0 +1,405 @@
+            TimeSpan retryDuration, 
+            ClientOperationMetricsBuilder clientOperationMetricsBuilder,
+            CancellationToken cancellationToken)
+        {
+            var sut = RpcCallExecutorFactory.Create(retryDuration, tentacleClientObserver);
+
+            sut.Execute(
+                RpcCallName,
+                action,
+                clientOperationMetricsBuilder,
+                cancellationToken);
+        }
+
+        private static Guid DelayFor100MillisecondsAction(CancellationToken cancellationToken)
+        {
+            Task.Delay(TimeSpan.FromMilliseconds(100), cancellationToken).Wait(cancellationToken);
+
+            return Guid.NewGuid();
+        }
+        
+        private static void ThenRpcCallMetricsShouldBeSuccessful(RpcCallMetrics metric, TimeSpan expectedRetryDuration, bool expectedWithRetries)
+        {
+            metric.Succeeded.Should().BeTrue();
+            metric.Exception.Should().BeNull();
+            metric.HasException.Should().BeFalse();
+            metric.WasCancelled.Should().BeFalse();
+
+            metric.AttemptsSucceeded.Should().BeTrue();
+
+            metric.End.Should().BeAfter(metric.Start);
+            metric.Duration.Should().Be(metric.End - metric.Start);
+
+            metric.RetryTimeout.Should().Be(expectedRetryDuration);
+            metric.RpcCallName.Should().Be(RpcCallName);
+            metric.WithRetries.Should().Be(expectedWithRetries);
+        }
+
+        private static void ThenRpcCallMetricsShouldBeFailed(RpcCallMetrics metric, TimeSpan expectedRetryDuration, Exception expectedException, bool expectedWithRetries)
+        {
+            metric.Succeeded.Should().BeFalse();
+            metric.Exception.Should().BeEquivalentTo(expectedException);
+            metric.HasException.Should().BeTrue();
+            metric.WasCancelled.Should().BeFalse();
+
+            metric.AttemptsSucceeded.Should().BeFalse();
+
+            metric.End.Should().BeAfter(metric.Start);
+            metric.Duration.Should().Be(metric.End - metric.Start);
+
+            metric.RetryTimeout.Should().Be(expectedRetryDuration);
+            metric.RpcCallName.Should().Be(RpcCallName);
+            metric.WithRetries.Should().Be(expectedWithRetries);
+        }
+
+        private static void ThenAttemptShouldBeSuccessful(TimedOperation attempt)
+        {
+            attempt.Succeeded.Should().BeTrue();
+            attempt.Exception.Should().BeNull();
+            attempt.WasCancelled.Should().BeFalse();
+
+            attempt.End.Should().BeAfter(attempt.Start);
+            attempt.Duration.Should().Be(attempt.End - attempt.Start);
+        }
+
+        private static void ThenAttemptShouldHaveFailed(TimedOperation attempt, Exception expectedException)
+        {
+            attempt.Succeeded.Should().BeFalse();
+            attempt.Exception.Should().BeEquivalentTo(expectedException);
+            attempt.WasCancelled.Should().BeFalse();
```

Comment:
Doesn't look like cancellation and WasCancelled is ever tested

Reply:
Is it not tested in `ExecuteWithRetries_MetricsShouldBeFailed_WhenCancellationTokenIsCancelled`? 

Or am I misunderstanding something?

Reply:
No, I missed it ;) Nothing to see here!!